src/timetable_cmd.cpp

```diff
@@ -221,19 +221,28 @@ CommandCost CmdSetVehicleOnTime(DoCommandFlag flags, VehicleID veh, bool apply_t
 	Vehicle *v = Vehicle::GetIfValid(veh);
 	if (v == nullptr || !v->IsPrimaryVehicle() || v->orders == nullptr) return CMD_ERROR;
 
+	/* A vehicle can't be late if its timetable hasn't started. */
+	if (!HasBit(v->vehicle_flags, VF_TIMETABLE_STARTED)) return CommandCost(STR_ERROR_TIMETABLE_NOT_STARTED);
```

Comment:
Feels like this should be conditional upon `apply_to_group`. If that is set then it probably doesn't matter if the first vehicle hasn't started, some of the others in the group may have?